### ISSUE TITLE
What "enabling support for ECMAScript 6" really means?

### ISSUE THREAD

**Issue by eTorAken** [eslint/eslint#4539]
Sorry for this question, but we are using ES6 in our code (and using `for of` statements) but when I set the `ecmaFeatures.forOf` to `false`, nothing seems to change.

We are using eslint@1.10.1

Could someone enlight me please?

**Comment by eslintbot** [comment]
@eTorAken Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
ecma Features enable individual features for the language. `forOf` enables the `for-of` loop (when false, using `for-of` is a syntax error. See http://eslint.org/docs/user-guide/configuring.html#specifying-language-options

**Comment by eTorAken** [comment]
Thanks for your answer @nzakas but it seems that `ecmaFeatures.forOf` set to `false` does not throw any syntax error.

I'm using this code: 
```js
const tests = {
  a: 'a',
  b: 'b',
};
for(const test of tests) {
  console.log(test);
}
```

Running `eslint` (v1.10.1) does not throw anything here.

**Comment by nzakas** [comment]
You probably are using either the es6 environment or babel-eslint. Both will override that setting.

**Comment by eTorAken** [comment]
You are right. TY

**Comment by eTorAken** [comment]
By the way, this may be a new rule to include.

**Comment by platinumazure** [comment]
Try no-restricted-syntax with a node type of ForOfStatement. :-)
